src/swarm-engine/Swarm/Game/Step/Const.hs

```diff
@@ -1214,6 +1214,30 @@ execConst runChildProg c vs s k = do
         Nothing -> raise Read ["Could not read", showT txt, "at type", prettyText ty]
         Just v -> return (mkReturn v)
       _ -> badConst
+    Print -> case vs of
+      [VText txt] -> do
+        paper <- ensureItem "paper" "print"
+        let newEntityName = "paper: " <> txt
+        robotInventory %= delete paper
+        robotInventory %= insert (paper & entityName .~ newEntityName)
+        return $ mkReturn newEntityName
+      _ -> badConst
+    Erase -> case vs of
+      [VText paperName] -> do
+        toErase <- ensureItem paperName "erase"
+        (paperName /= "paper")
+          `holdsOrFail` ["That is already blank!"]
+        ("paper: " `T.isPrefixOf` paperName)
```

Comment:
It seems fragile in some way to use an "in-band" indicator for erasability and printability, in that it must be inferred by the entity's name.  Perhaps instead we want to define a "printable" property?  Then we could have different printable entities like "papyrus", "clay", "whiteboard".  Each of them can still be subject to the name transformation with the colon.  A property called "printable" would be inherent to the "unmarked" entity, whereas the property called "erasable" would be dynamically set once the entity undergoes a "print".

Reply:
Oh, that's a nice idea.  Let me play with that.

Reply:
@kostmo What do you think?  I decided to just go with a single property `printable` that determines whether something can be both printed and erased.  Having a separate property `erasable` didn't seem to make sense to me since it's not something that would make sense to set on base entities, and the only reason to `erase` something is that you previously `print`ed on it.

Reply:
I guess this does mean that you can still `print` on something that has been previously `print`ed on, like this:
```
> print "paper" "3"
it0 : Text = "paper: 3"
> print "paper: 3" "7"
it1 : Text = "paper: 3: 7"
```
That's a bit strange, perhaps, but I don't think it's really a problem.